src/agent/debuglet.ts

```diff
@@ -107,6 +113,34 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise stores a promise. This promise is initially null and can be
+ * assigned to a new promise when refresh is called. This promise will be
+ * resolved by calling member function resolve.
+ */
+export class CachedPromise {
+  private promise: Promise<void>|null;
+  private promiseResolve: () => void;
+
+  constructor() {
```

Comment:
nit: You could just have `private promise: Promise<void>|null = null` and then not have a constructor at all.

Reply:
Done.